src/frontend/src/binder/insert.rs

```diff
@@ -312,10 +314,22 @@ impl Binder {
         let msg = match expected_types.len().cmp(&exprs.len()) {
             std::cmp::Ordering::Less => "INSERT has more expressions than target columns",
             _ => {
+                let expr_len = exprs.len();
                 return exprs
                     .into_iter()
                     .zip_eq_fast(expected_types.iter().take(expr_num))
-                    .map(|(e, t)| e.cast_assign(t.clone()).map_err(Into::into))
+                    .enumerate()
+                    .map(|(i, (e, t))| {
+                        let res = e.cast_assign(t.clone());
+                        if expr_len > 1 {
```

Comment:
No need to check this?

Reply:
Looks verbose to me for the case with 1 col 🫢